mods/cnc/maps/gdi10a/gdi10a.lua

```diff
@@ -0,0 +1,413 @@
+	if #prodParms.AttackGrp == 0 then
+		-- Choose a group to build
+		prodParms.ModelGroup = Utils.Random(prodParms.ModelGroups)
+	else
+		RemoveLostActors(prodParms.AttackGrp)
+	end
+	-- Which units do we not have yet?  Randomly build one of them.
+	local haveTypes = {}
+	local s = string.format('ProduceUnit(%s,_): Have', factory.Type)
+	for i, actor in ipairs(prodParms.AttackGrp) do
+		table.insert(haveTypes, actor.Type)
+		s = s .. ' ' .. actor.Type
+	end
+	local needUnits = compareTables(haveTypes, prodParms.ModelGroup)
+	s = s .. ', Need'
+	for i, actor in ipairs(needUnits) do
+		s = s .. ' ' .. actor
+	end
+
+	if #needUnits > 0 then
+		-- Continue building the group
+		local toBuild = { Utils.Random(needUnits) }
+		print(string.format('; Build %s', toBuild[1]))
+		-- TODO What if Build returns false? Do we try later?
+		factory.Build(toBuild, function(unit)
+			-- Unit ready
+			for _,v in ipairs(unit) do
+				table.insert(prodParms.AttackGrp, v)
+			end
+			local delay = Utils.RandomInteger(DateTime.Seconds(12), DateTime.Seconds(17))
+			Trigger.AfterDelay(delay, function()
+				ProduceUnit(factory, prodParms)
+			end)
+		end)
+	else
+		-- Group ready
+		local path = Utils.Random(prodParms.AttackPaths)
+		-- Tried to use campain.lua: MoveAndHunt but it wants path elements to have a Location property
+		SendUnits(prodParms.AttackGrp, path)
+		prodParms.AttackGrp = {}
+		prodParms.ModelGroup = nil
+		Trigger.AfterDelay(prodParms.Cooldown, function()
+			ProduceUnit(factory, prodParms)
+		end)
+		s = s .. '; Attack!'
+	end
+
+	Media.Debug(s)
+	print(s)
+end
+
+ProduceHarvester = function(factory, prodParms)
+	local harv_type = "harv"
+	if BankBalance(Nod) < Actor.Cost(harv_type) then
+		-- Try again later
+		Media.Debug(string.format('ProduceHarvester: Have $%d, need $%d', BankBalance(Nod), Actor.Cost(harv_type)))
+		Trigger.AfterDelay(DateTime.Seconds(5), function()
+			ProduceUnit(factory, prodParms)
+		end)
+	else
+		BuildingHarvester = true
+		factory.Build({ harv_type }, function(units)
+			-- Unit ready
+			for idx, unit in pairs(units) do
+				if unit.Type == 'harv' then
+					s='Harvester built: ' .. ActorString(unit)
+					Media.Debug(s)
+					print(s)
+					--Unnecessary:
+					--unit.FindResources()
+					BuildingHarvester = false
+				end
+			end
+			-- Continue building armour/vehicles
+			Trigger.AfterDelay(DateTime.Seconds(5), function()
+				ProduceUnit(factory, prodParms)
+			end)
+		end)
+	end
+end
+
+GetNodCyard = function()
+	local cyards = Nod.GetActorsByType("fact")
+	if (#cyards < 1) then
+		return nil -- Lost cyard; can never build again
+	else
+		return cyards[1]
+	end
+end
+
+-- Remove captured and killed actors from a table
+RemoveLostActors = function (tbl)
+    for i = #tbl, 1, -1 do
+        if tbl[i].IsDead or tbl[i].Owner ~= Nod then
+            table.remove(tbl, i)
+        end
+    end
+end
+
+
+Tick = function()
+	-- Check objectives
+	if DateTime.GameTime > DateTime.Seconds(5) then
```

Comment:
Yes this was partly based on a resurrected PR branch.  So what do you want to do?I've tested this map dozens of times.    This works just fine AFAIK.

Reply:
Mostly just wanted clarity if reinforcement is on the to-do or not. I'd guess not since the terrain is a bit more awkward for it than `gdi-10b` would be.